lib/src/components/NxCombobox/types.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import * as PropTypes from 'prop-types';
+import { omit } from 'ramda';
+import { HTMLAttributes, ReactNode } from 'react';
+import { PublicProps as NxTextInputProps, propTypes as nxTextInputPropTypes } from '../NxTextInput/types';
+
+import DataItem from '../../util/DataItem';
+
+type InputProps = Omit<NxTextInputProps, 'type' | 'value' | 'isPristine' | 'readOnly'>;
+
+export interface Props<T extends string | number = string> extends Omit<HTMLAttributes<HTMLDivElement>, 'onSelect'> {
+  searchText: string;
+  loading?: boolean | null;
+  error?: ReactNode;
+  matches: DataItem<T>[];
+  onSelect: (m: DataItem<T>) => void;
```

Comment:
Should this be exposed externally? I'd think that from an external perspective, the user selecting from the list would just appear as a change to the value of the component (e.g. something communicated by `onChange`)

Reply:
Agreed with you, it just changed the input text value once an item is selected, which can be updated through `onChange`. Updated here 5088cbfd68f100b47e5c21f4bef3975e6b74e018.